src/webgpu/api/validation/validation_test.ts

```diff
@@ -280,15 +281,41 @@ export class ValidationTest extends GPUTest {
     }
   }
 
-  expectValidationError(fn: Function, shouldError: boolean = true): void {
+  async expectValidationErrorImmediate<T>(
+    fn: () => T | Promise<T>,
```

Comment:
Isn't `T` sufficient as it can be a `Promise<T>` as well?

Reply:
`T` is also used in the return type `Promise<T>`. They input should be flexible and the output should not be.

Reply:
I guess doing this makes the type inference ambiguous, but somehow it works out and doesn't end up with `Promise<Promise<void>>` at the end.

Reply:
That's because if the return value is already a promise, an async function returns it directly. It only wraps the result if it's not a Promise